sql/src/main/java/io/crate/planner/consumer/QuerySplitter.java

```diff
@@ -205,14 +209,16 @@ public Symbol visitFunction(Function function, InternalSplitterCtx context) {
                 if (AndOperator.NAME.equals(function.info().ident().name())) {
                     List<Symbol> newArgs = new ArrayList<>(function.arguments().size());
                     for (Symbol argument : function.arguments()) {
-                        RelationCount argumentCount = context.countMap.get(argument);
-                        if (argumentCount.numOther == 0) {
-                            context.relationQueryParts.add(argument);
-                            argument = Literal.newLiteral(true);
-                        } else {
-                            context.insideFunction = true;
-                            argument = process(argument, context);
-                            context.insideFunction = false;
+                        if (!argument.symbolType().isValueSymbol()) {
+                            RelationCount argumentCount = context.countMap.get(argument);
```

Comment:
if it is null here the RelationCountVisitor did something wrong, should I add an assertion?